### DISCUSSION TITLE
Compile FPrime for Google Dev Board Mini

### DISCUSSION THREAD

**Discussion by JHC0de** [nasa/fprime#1997] [Q&A] [answered]
Is it possible to compile for Google dev board mini if so does anyone have an example tool chain. Looking for alternatives to RPI… other board suggestions are also welcome ;)

**Comment by capsulecorplab** [comment]
If you're looking for an alternative to RPI, I managed to cross compile the MathComponent as of fprime v3.0.0 for the BeagleBone https://github.com/nasa/fprime/discussions/1437

**Reply by JHC0de** [reply]
Awesome, Thanks. Ill take a look into that, Is the beagle bone powerful enough or did it cope fine running FPrime?

**Reply by SterlingPeet** [reply]
I've run F Prime on much smaller and much less powerful platforms.  However, F Prime is a framework, so it will depend on how you use it.  The SystemReference deployment with all of its hardware bits and pieces running is pretty tame for that platform.  I would imagine there are complex GNC algorithms or vision processing algorithms that would make it struggle, although not as a result of using F Prime.

**Comment by thomas-bc** [comment]
We have also been running on ARM64 Odroids using the latest F´ 3.2.0. You can find the project here: https://github.com/fprime-community/fprime-workshop-led-blinker
This was used in a very recent workshop so you can find extensive step-by-step instructions and explanations in the [docs/](https://github.com/fprime-community/fprime-workshop-led-blinker/tree/main/docs) folder!

**Reply by JHC0de** [reply]
Okay awesome Thanks! What are your thoughts on this being adapted to work on other board/How hard is it to build a tool chain for other boards?

**Reply by LeStarch** [reply]
Cross-compiling can be a bit of a challenge (in general regardless of F´ usage).  It really depends on:

1. Target operating system or lack there-of
2. Vendor supplied tools and environment
3. Complexity of the system you wish to build

e.g.  running on some flavor of embedded linux is fairly easy.  Wrap the appropriate `gcc` tools and you are done, as seen in the arm package above.  Running full F´ without an Os on a chip with vendor-supplied non-gcc tools is a different question.

Do you have a target platform in mind?  We can provide more details if we know generally what environment we are dealing with.

**Reply by JHC0de** [reply]
There are two implementations i'd like to try:
1. Id like to try Google dev board mini just running any flavour of Embedded Linux that will work best for it.
2. (Know this one will be much harder) Using the NXP mpc5748g to do sensor data collection and passing data to a port for use of the Google board.

Is this enough information? what else would you need to know and i can try and provide (Relativly new to this (Grad working in automotive, Have used AUTOSAR which is a similar concept to FPrime))

**Reply by JHC0de** [reply]
Also intrested in an option using the Snickerdoodle Black FPGA and then running two Google Dev Board Minis for redundancy. Similar to the Perseverance architecture (atleast from what ive read around it in the FPrime Docs)

**Reply by LeStarch** [reply, marked as answer]
Both the processors on the dev board seem to be arm, so with a sufficiently new version of Linux installed, F´ should just work when using the arm-linux cross compilers.

**Reply by LeStarch** [reply]
The NXP would be more tricky, but if NXP provides gcc compilers it should be doable.
